File: src/main/java/net/hectus/neobb/turn/person_game/block/PTVerdantFroglight.java
package net.hectus.neobb.turn.person_game.block;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.turn.Turn;
import net.hectus.neobb.turn.default_game.block.BlockTurn;
import net.hectus.neobb.turn.person_game.categorization.WinConCategory;
import org.bukkit.block.Block;

import java.util.List;

public class PTVerdantFroglight extends BlockTurn implements WinConCategory {
    public PTVerdantFroglight(NeoPlayer player) { super(player); }
    public PTVerdantFroglight(Block data, NeoPlayer player) { super(data, player); }

    @Override
    public double damage() {
        return 4;
    }

    @Override
    public List<Class<? extends Turn<?>>> counteredBy() {
        return List.of(PTFenceGate.class);
    }

    @Override
    public boolean unusable() {
        return player.nextPlayer().health() > 4.0;
    }
}

File: src/main/kotlin/net/hectus/neobb/modes/turn/person_game/block/PTVerdantFroglight.kt
package net.hectus.neobb.modes.turn.person_game.block

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.modes.turn.Turn
import net.hectus.neobb.modes.turn.default_game.block.BlockTurn
import net.hectus.neobb.modes.turn.person_game.categorization.WinConCategory
import net.hectus.neobb.player.NeoPlayer
import org.bukkit.block.Block
import kotlin.reflect.KClass

class PTVerdantFroglight(data: Block?, cord: Cord?, player: NeoPlayer?) : BlockTurn(data, cord, player), WinConCategory {
    override val damage: Double = 4.0

    override fun counteredBy(): List<KClass<out Turn<*>>> {
        return listOf(PTFenceGate::class)
    }

    override fun unusable(): Boolean {
        return player!!.nextPlayer().health > 4.0
    }
}
